test-templates/runtime-service/src/lib.rs

```diff
@@ -74,6 +74,38 @@ impl Guest for Component {
         oplog_commit(replicas);
         println!("Finished commit");
     }
+
+    fn atomic_region() {
+        let now = std::time::SystemTime::now();
+        println!("Starting atomic region at {now:?}");
+
+        let begin = mark_begin_operation();
+
+        remote_side_effect("1"); // repeated 3x
```

Comment:
repeated 3x here means?

Reply:
there is a failure between side effect "2" and "3" which is controlled from the outside in a way that first two times it fails, third time it is not. Because this side effect is between begin/end operation markers, it is going to be performed three times, but not more, because after the third the region is committed.